File: XXRootViewController.m
#import "XXRootViewController.h"
#import <spawn.h>
NSString *imageToFloodURL;

@implementation XXRootViewController
- (void)loadView
{
    [super loadView];
    self.view.backgroundColor = [UIColor blackColor];
    // start creating the buttons (this is basic as fuck, but if you want to use it obviously you would make it look better)
    UIButton *librarybutton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [librarybutton addTarget:self action:@selector(chooseFromLibrary:) forControlEvents:UIControlEventTouchUpInside];
    [librarybutton setTitle:@"Choose from library" forState:UIControlStateNormal];
    librarybutton.frame = CGRectMake(20, 80, self.view.frame.size.width - 40, 40);
    [self.view addSubview:librarybutton];
    
    UIButton *urlbutton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [urlbutton addTarget:self action:@selector(chooseFromURL:) forControlEvents:UIControlEventTouchUpInside];
    [urlbutton setTitle:@"Choose from URL" forState:UIControlStateNormal];
    urlbutton.frame = CGRectMake(20, 140, self.view.frame.size.width - 40, 40);
    [self.view addSubview:urlbutton];
    
    UIButton *floodbutton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [floodbutton addTarget:self action:@selector(startFlooding:) forControlEvents:UIControlEventTouchUpInside];
    [floodbutton setTitle:@"Start flooding" forState:UIControlStateNormal];
    floodbutton.frame = CGRectMake(20, 200, self.view.frame.size.width - 40, 40);
    [self.view addSubview:floodbutton];

    UIButton *stopfloodbutton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [stopfloodbutton addTarget:self action:@selector(stopFlooding:) forControlEvents:UIControlEventTouchUpInside];
    [stopfloodbutton setTitle:@"Stop flooding" forState:UIControlStateNormal];
    stopfloodbutton.frame = CGRectMake(20, 200, self.view.frame.size.width - 40, 40);
    [self.view addSubview:stopfloodbutton];
}
-(void)chooseFromLibrary:(id)sender
{
    // This code presents a prompt for an image from the phone's library
    UIImagePickerController* imagePicker = [[UIImagePickerController alloc]init];
    imagePicker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
    imagePicker.delegate = self;
    [self presentViewController:imagePicker animated:true completion:nil];
}

// this is the method called when the picker finishes
- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info
{
    [picker dismissViewControllerAnimated:YES completion:nil];
    // saves the selected image as /User/Documents/floodimage.png
    [UIImagePNGRepresentation([info objectForKey:UIImagePickerControllerOriginalImage]) writeToFile:@"/User/Documents/floodimage.png" atomically:YES];
    imageToFloodURL = @"file:///User/Documents/floodimage.png";
}
-(void)chooseFromURL:(id)sender
{
    // create an alert controller to handle getting the url
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle: @"TrollDrop" message: @"Choose URL" preferredStyle:UIAlertControllerStyleAlert];
    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField)
    {
        // configure the text input field we just added
        textField.placeholder = @"Name";
        textField.textColor = [UIColor blackColor];
        textField.clearButtonMode = UITextFieldViewModeWhileEditing;
        textField.borderStyle = UITextBorderStyleRoundedRect;
    }];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleDefault handler:^(UIAlertAction *action){}]];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Okay" style:UIAlertActionStyleDefault handler:^(UIAlertAction *action)
    {
        NSArray *textfields = alertController.textFields;
        UITextField *imageurl = textfields[0];
        imageToFloodURL = imageurl.text;
    }]];
    // show the alert
    [self presentViewController:alertController animated:YES completion:nil];
}
-(void)startFlooding:(id)sender
{
    /* here you should put your objective c code for trolldrop











    */
}
-(void)stopFlooding:(id)sender
{
    // put stuff here to stop flooding
}
@end

File: XXRootViewController.swift
import UIKit

class XXRootViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {

    var imageToFloodURL = ""
    var logArea: UITextView?

    func trollLog(_ message: String) {
        logArea?.text = (logArea?.text ?? "") + message + "\n"
    }

    override func loadView() {
        super.loadView()
        view.backgroundColor = UIColor.black
        // start creating the buttons
        let librarybutton = UIButton(type: .roundedRect)
        librarybutton.addTarget(self, action: #selector(XXRootViewController.choose(fromLibrary:)), for: .touchUpInside)
        librarybutton.setTitle("Choose from library", for: .normal)
        librarybutton.frame = CGRect(x: 20, y: 80, width: view.frame.size.width - 40, height: 40)
        view.addSubview(librarybutton)

        let urlbutton = UIButton(type: .roundedRect)
        urlbutton.addTarget(self, action: #selector(XXRootViewController.choose(fromURL:)), for: .touchUpInside)
        urlbutton.setTitle("Choose from URL", for: .normal)
        urlbutton.frame = CGRect(x: 20, y: 120, width: view.frame.size.width - 40, height: 40)
        view.addSubview(urlbutton)

        let floodbutton = UIButton(type: .roundedRect)
        floodbutton.addTarget(self, action: #selector(XXRootViewController.startFlooding(_:)), for: .touchUpInside)
        floodbutton.setTitle("Start flooding", for: .normal)
        floodbutton.frame = CGRect(x: 20, y: 160, width: view.frame.size.width - 40, height: 40)
        view.addSubview(floodbutton)

        let stopfloodbutton = UIButton(type: .roundedRect)
        stopfloodbutton.addTarget(self, action: #selector(XXRootViewController.stopFlooding(_:)), for: .touchUpInside)
        stopfloodbutton.setTitle("Stop flooding", for: .normal)
        stopfloodbutton.frame = CGRect(x: 20, y: 200, width: view.frame.size.width - 40, height: 40)
        view.addSubview(stopfloodbutton)

        logArea = UITextView(frame: CGRect(x: 20, y: 240, width: view.frame.size.width - 40, height: 100))
        logArea!.isEditable = false
        logArea!.text = "-trolldropapp log-\n"
        view.addSubview(logArea!)
    }

    @objc func choose(fromLibrary sender: Any?) {
        // This code presents a prompt for an image from the phone's library
        let imagePicker = UIImagePickerController()
        imagePicker.sourceType = .photoLibrary
        imagePicker.delegate = self
        present(imagePicker, animated: true)
    }

    // this is the method called when the picker finishes
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String: Any]) {
        picker.dismiss(animated: true)

        let directory = NSTemporaryDirectory()
        let fileName = "image.png"
        let fullURL = URL(fileURLWithPath: fileName, relativeTo: URL(fileURLWithPath: directory));
        trollLog("image saved to " + fullURL.absoluteString)

        try! UIImagePNGRepresentation(info[UIImagePickerControllerOriginalImage] as! UIImage)!.write(to: fullURL)
        imageToFloodURL = "file:///tmp/floodimage.png"
    }

    @objc func choose(fromURL sender: Any?) {
        // create an alert controller to handle getting the url
        let alertController = UIAlertController(title: "TrollDrop", message: "Choose URL", preferredStyle: .alert)
        alertController.addTextField(configurationHandler: { textField in
            // configure the text input field we just added
            textField.placeholder = "Name"
            textField.textColor = UIColor.black
            textField.clearButtonMode = .whileEditing
            textField.borderStyle = .roundedRect
        })
        alertController.addAction(UIAlertAction(title: "Cancel", style: .default, handler: { action in
        }))
        alertController.addAction(UIAlertAction(title: "Okay", style: .default, handler: { action in
            let textfields = alertController.textFields
            let imageurl: UITextField = textfields![0]
            self.imageToFloodURL = imageurl.text!
        }))
        // show the alert
        present(alertController, animated: true)
    }

    @objc func startFlooding(_ sender: Any?) {
        // here you should put your objective c code for trolldrop
        trollLog("start flooding")

    }

    @objc func stopFlooding(_ sender: Any?) {
        trollLog("stop flooding")

    }

}